### DISCUSSION TITLE
a confusing case in `selection` or `collapsed selection`

### DISCUSSION THREAD

**Discussion by acumen1005** [ianstormtaylor/slate#5623] [General]
I'm confused by this example. Consider the following Slate document

`[{text: 'one '}, {text: 'two', bold: true}, {text: ' three'}]`

It would render like so:

> one two three

Now if we selected the "two":

> one |**two**| three

Then move the focus handle to left until they(anchor and focus) collapse, like:

> one |two three

Now the selection is:

```
{
  anchor: { path: [1], offset: 0 },
  focus: { path: [0], offset: 5 }
}
```

My confusion arises in how to determine if the selection has collapsed, given that the anchor and focus in the above selection do not definitely match.

**Comment by gerardnico** [comment]
The selection has not collapsed as the anchor and the focus are not on the same point.

A collapsed selection would have the anchor and the focus on the same point.

Example:
with an editor like that where cursor is a collapsed selection (ie range ie cursor)
```javascript
editor = <editor>
            <paragraph>
                <text>
                    the text
                    <cursor/>
                </text>
            </paragraph>
  </editor>
```
The selection would be
```javascript
editor.selection == {
            anchor: {path: [0, 0], offset: 8},
            focus: {path: [0, 0], offset: 8}
}
```
Then
```javascript
SlateRange.isCollapsed(editor.selection) === true
```

The cursor is at :
* 0: because paragraph is the first element in the hierarchy path
* 0: because text is the first text node in the element
* offset: 8, the 8st character

**Comment by rahulbansal16** [comment]
{
  anchor: { path: [1], offset: 0 },
  focus: { path: [0], offset: 5 }
}
Why is focus 0th node? Shouldn't it be the first node as you are navigating inside the node?
